src/explorer/WebAppProvider.ts

```diff
@@ -60,31 +62,72 @@ export class WebAppProvider extends SubscriptionTreeItem {
+
+        await setAppWizardContextDefault(wizardContext);
+
+        const promptSteps: AzureWizardPromptStep<IAppServiceWizardContext>[] = [];
+        const executeSteps: AzureWizardExecuteStep<IAppServiceWizardContext>[] = [];
+
+        promptSteps.push(new SiteNameStep());
+
         const workspaceConfig: WorkspaceConfiguration = workspace.getConfiguration(extensionPrefix);
         const advancedCreation: boolean | undefined = workspaceConfig.get(configurationSettings.advancedCreation);
-        let newSite: Site | undefined;
-        try {
-            newSite = await createWebApp(actionContext, this.root, { advancedCreation }, showCreatingTreeItem);
-        } catch (error) {
-            if (!parseError(error).isUserCancelledError && !advancedCreation) {
-                const message: string = `Modify the setting "${extensionPrefix}.${configurationSettings.advancedCreation}" if you want to change the default values when creating a Web App in Azure.`;
-                const btn: MessageItem = { title: 'Turn on advanced creation' };
-                // tslint:disable-next-line: no-floating-promises
-                ext.ui.showWarningMessage(message, btn).then(async result => {
-                    if (result === btn) {
-                        const projectConfiguration: WorkspaceConfiguration = workspace.getConfiguration('appService');
-                        await projectConfiguration.update('advancedCreation', true, ConfigurationTarget.Global);
-                    }
-                });
-            }
-            throw error;
-        }
-        if (newSite === undefined) {
-            throw new UserCancelledError();
+        if (advancedCreation) {
+            promptSteps.push(new ResourceGroupListStep());
+            promptSteps.push(new SiteOSStep());
+            promptSteps.push(new SiteRuntimeStep());
+            promptSteps.push(new AppServicePlanListStep());
         } else {
-            const siteClient: SiteClient = new SiteClient(newSite, this.root);
-            return new WebAppTreeItem(this, siteClient);
+            promptSteps.push(new SiteOSStep()); // will be skipped if there is a smart default
+            promptSteps.push(new SiteRuntimeStep());
+            executeSteps.push(new ResourceGroupCreateStep());
+            executeSteps.push(new AppServicePlanCreateStep());
+        }
+        executeSteps.push(new SiteCreateStep());
+
+        if (wizardContext.newSiteOS !== undefined) {
+            SiteOSStep.setLocationsTask(wizardContext);
         }
+
+        const title: string = 'Create new web app';
+        const wizard: AzureWizard<IAppServiceWizardContext> = new AzureWizard(wizardContext, { promptSteps, executeSteps, title });
+
+        await wizard.prompt(actionContext);
+        // tslint:disable-next-line strict-boolean-expressions
+        if (showCreatingTreeItem) {
+            // tslint:disable-next-line no-non-null-assertion
+            showCreatingTreeItem(wizardContext.newSiteName!);
```

Comment:
Doesn't have to be this PR, but eventually this repo should start using `nonNullProp`, etc. like other repos